lib/src/services/biometric_service/method_channel_biometric_service.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:flutter/services.dart' show MethodChannel, PlatformException;
+import 'package:gigya_flutter_plugin/gigya_flutter_plugin.dart';
+
+import '../../models/enums/methods.dart';
+import 'biometric_service.dart';
+
+/// This class represents a [BiometricService] that uses a [MethodChannel]
+/// for its implementation.
+class MethodChannelBiometricService extends BiometricService {
+  /// The default constructor.
+  const MethodChannelBiometricService(this._channel);
+
+  final MethodChannel _channel;
+
+  @override
+  Future<bool> isAvailable() async {
+    try {
+      final bool? result = await _channel.invokeMethod<bool>(
+        BiometricMethods.isAvailable.methodName,
+      );
+
+      return result ?? false;
+    } on PlatformException catch (exception) {
+      throw GigyaError.fromPlatformException(exception);
+    }
+  }
+
+  @override
+  Future<bool> isLocked() async {
+    try {
+      final bool? result = await _channel.invokeMethod<bool>(
+        BiometricMethods.isLocked.methodName,
+      );
+
+      return result ?? false;
+    } on PlatformException catch (exception) {
+      throw GigyaError.fromPlatformException(exception);
+    }
+  }
+
+  @override
+  Future<bool> isOptIn() async {
+    try {
+      final bool? result = await _channel.invokeMethod<bool>(
+        BiometricMethods.isOptIn.methodName,
+      );
+
+      return result ?? false;
+    } on PlatformException catch (exception) {
+      throw GigyaError.fromPlatformException(exception);
+    }
+  }
+
+  @override
+  Future<bool> optIn({
```

Comment:
The methods `optIn`, `optOut`, `lockSession` and `unlockSession` are all typed as returning a `Future<void>` in the biometric service interface. 

If the method succeeds, it does not return a result (hence the `success(null/nil)` on the native side).

If the method fails, it throws a `PlatformException`, which you already handled here, by converting it into a `GigyaError`

So you'll have to change these methods to do the following:
- `invokeMethod<void>` instead of `invokeMethod<bool>`
- the return type is `Future<void>` instead of `Future<bool>`